gmail/gmail-connector/gmail-connector-2.0.0/org.wso2.carbon.connector/src/main/java/org/wso2/carbon/connector/GmailUtils.java

```diff
@@ -0,0 +1,836 @@
+                        log.info("Cannot retrive \"Message Content\".");
+                    }
+
+                    if (attachmentContentIDs.length() > 0) {
+                        OMElement attachments = factory.createOMElement("attachemnts", ns);
+                        messageElement.addChild(attachments);
+                        attachmentContentIDs.setLength(attachmentContentIDs.length() - 1);
+                        attachments.setText(attachmentContentIDs.toString());
+                    }
+                }
+            }
+        }
+        messageContext.getEnvelope().getBody().addChild(result);
+    }
+
+    /**
+     * Process message body.
+     *
+     * @param message
+     *            Message to be processed.
+     * @param messageContext
+     *            Message Context
+     * @param attachmentContentIDs
+     *            file names as the content IDs of the attachments
+     * @param messageID
+     *            ID of the message.
+     * @return
+     * @throws IOException
+     * @throws MessagingException
+     */
+    private static String processMessageBody(Message message, MessageContext messageContext,
+                                             StringBuilder attachmentContentIDs, String messageID)
+            throws IOException,
+            MessagingException {
+        Object content = message.getContent();
+        if (content instanceof Multipart) {
+            Multipart multiPart = (Multipart) content;
+            StringBuilder builder = new StringBuilder();
+            return procesMultiPart(builder, multiPart, messageContext, attachmentContentIDs,
+                    messageID);
+        } else if (content instanceof String) {
+            return content.toString();
+        } else if (content instanceof InputStream) {
+            InputStream inStream = (InputStream) content;
+            BufferedReader br = new BufferedReader(new InputStreamReader(inStream));
+            String line;
+            StringBuilder builder = new StringBuilder();
+            while ((line = br.readLine()) != null) {
+                builder.append(line);
+            }
+            br.close();
+            inStream.close();
+            return builder.toString();
+        }
+        log.error("invalid message content");
+        return null;
+    }
+
+    /**
+     * Reads the batch number from the input string.
+     *
+     * @param batchString
+     *            input string
+     * @return batch number
+     * @throws NumberFormatException
+     *             if the batch number is not an integer
+     * @throws ConnectException
+     *             if the batch number is not a positive integer
+     */
+    public static int getBatchNumber(String batchString) throws NumberFormatException,
+            ConnectException {
+        int batchNumber;
+        if (batchString != null && !"".equals(batchString.trim())) {
+            batchNumber = Integer.parseInt(batchString);
```

Comment:
Please catch NumberFormatException while converting String to Long, Integer, Float and Double.